### PR TITLE
Pipeline, Polina Zhilina - 21FPL2

### PR DESCRIPTION
**Pull request by polyamba**

### CODE REVIEW

**Review by marina-kaz — COMMENTED** [review]
No changes concerning the pipeline to review

**Review by marina-kaz — COMMENTED** [review]
No changes to review

File: lab_6_pipeline/pipeline.py [hunk 1]
```diff
@@ -6,6 +6,30 @@
 
 from core_utils.article.article import SentenceProtocol
 from core_utils.article.ud import OpencorporaTagProtocol, TagConverter
+from core_utils.constants import ASSETS_PATH
+
+
+class FileNotFoundError(Exception):
```
Comment (marina-kaz) [inline lab_6_pipeline/pipeline.py]: This exception exists already

File: lab_6_pipeline/pipeline.py [hunk 2]
```diff
@@ -6,6 +6,30 @@
...
+    """
+    If file does not exist
+    """
+
+
+class NotADirectoryError(Exception):
```
Comment (marina-kaz) [inline lab_6_pipeline/pipeline.py]: This exception exists already

File: lab_6_pipeline/pipeline.py [hunk 3]
```diff
@@ -19,10 +43,22 @@ def __init__(self, path_to_raw_txt_data: Path):
         Initializes CorpusManager
         """
 
+        self._path = path_to_raw_txt_data
+        self._storage = []
```
Comment (marina-kaz) [inline lab_6_pipeline/pipeline.py]: Are you sure it is supposed to be a list?

File: lab_6_pipeline/pipeline.py [hunk 4]
```diff
@@ -19,10 +43,22 @@ def __init__(self, path_to_raw_txt_data: Path):
...
+
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
+        if not self._path.exists():
+            raise FileNotFoundError
+        if not self._path.is_dir():
+            raise NotADirectoryError
+        if len(meta_files) != len(raw_files):
```
Comment (marina-kaz) [inline lab_6_pipeline/pipeline.py:55]: Where do you define these variables?

**Review by marina-kaz — COMMENTED** [review]
Well, it is a start :)
Keep going!

File: lab_6_pipeline/pipeline.py (hunk 4 above)
Comment (polyamba) [reply lab_6_pipeline/pipeline.py:55]: i'm so sorry just saw this kind of check but did it badly
now I've done it in a better way

File: lab_6_pipeline/pipeline.py (hunk 1 above)
Comment (polyamba) [reply lab_6_pipeline/pipeline.py]: fixed

File: lab_6_pipeline/pipeline.py (hunk 2 above)
Comment (polyamba) [reply lab_6_pipeline/pipeline.py]: fixed

File: lab_6_pipeline/pipeline.py (hunk 3 above)
Comment (polyamba) [reply lab_6_pipeline/pipeline.py]: fixed

**Comment by polyamba** [comment]
I have a problem with PR name check, so how can I fix it if the work named as it was sad in instruction, but still this error exists

File: lab_6_pipeline/pipeline.py [hunk 5]
```diff
@@ -19,20 +34,48 @@ def __init__(self, path_to_raw_txt_data: Path):
...
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
+        if not self._path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+        if not self._path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError
+
+        meta_files = [meta for meta in self._path_to_raw_txt_data.glob(r'*_meta.json')]
```
Comment (marina-kaz) [inline lab_6_pipeline/pipeline.py]: For the mark 4, meta files are not expected, so it is better to remove working with them so as to let the tests pass

**Review by marina-kaz — COMMENTED** [review]
I see progress, keep going!

File: lab_6_pipeline/pipeline.py (hunk 5 above)
Comment (polyamba) [reply lab_6_pipeline/pipeline.py]: changed the score

**Pull request closed without merging**